### ISSUE TITLE
div or ul?

### ISSUE THREAD

**Issue by Ademsi** [daveaglick/FluentBootstrap#27]
Why do we get ("FluentBootstrap lets you output this"):
```html
<div class="navbar-left navbar-nav nav">
 <li>...</li>
 <li>...</li>
</div>
```

and not:
```html
<ul class="navbar-left navbar-nav nav">
 <li>...</li>
 <li>...</li>
</ul>
```

**Comment by daveaglick** [comment]
Good question - that certainly seems like a bug.

I can see where it's happening and all my unit tests also have the `div` instead of the `ul`. Not sure how that slipped by. Let me just take a look back through the commits to see if there's some reason why I would have done it intentionally (I can't imagine why). Even if I did mean to do it, it's not valid HTML5 markup so it needs to get fixed.

I'll have a patch out hopefully later today.

**Comment by Ademsi** [comment]
Thank you so much for your reply and for your splendid work :)

 

In fact I was particularly afraid to submit the message (I was thinking I was missing something :)).

 

On my box, I tried this small change in NavbarNav.cs:

 

namespace FluentBootstrap.Navbars

{

    public class NavbarNav : Tag, INavbarComponent,

        ICanCreate<NavbarLink>,

        ICanCreate<Dropdown>

    {

        internal NavbarNav(BootstrapHelper helper)

            : base(helper, "ul", Css.Nav, Css.NavbarNav, Css.NavbarLeft)  // ADEMSI 20150507

        {

        }

        

        protected override void OnStart(System.IO.TextWriter writer)

        {

            // Make sure we're in a collapse, but only if we're also in a navbar

            if (GetComponent<Navbar>() != null && GetComponent<NavbarCollapse>() == null)

            {

                GetHelper().NavbarCollapse().Component.Start(writer);

            }

 

            base.OnStart(writer);

        }

    }

}

 

 

I’ll use your so nice work with ASP.Net WebPages.

 

Bien cordialement,

 

Didier Pilois

Tél. 09 72 12 86 47



 



 

www.ademsi.com <http://www.ademsi.com/> 

Association N° W641005705

Régie par la Loi du 1er juillet 1901


Adresse de Gestion

C/O Didier Pilois
Contrada Ciancio 852/L
91025 Marsala TP
ITALIA

 

Siège social

64200 Biarritz

FRANCE

 

De : Dave Glick [mailto:notifications@github.com] 
Envoyé : jeudi 7 mai 2015 17:22
À : daveaglick/FluentBootstrap
Cc : Association ADEMSI
Objet : Re: [FluentBootstrap] div or ul? (#27)

 

Good question - that certainly seems like a bug.

I can see where it's happening and all my unit tests also have the div instead of the ul. Not sure how that slipped by. Let me just take a look back through the commits to see if there's some reason why I would have done it intentionally (I can't imagine why). Even if I did mean to do it, it's not valid HTML5 markup so it needs to get fixed.

I'll have a patch out hopefully later today.
